### PR TITLE
Add puzzle as an alternative browser

### PR DESCRIPTION
**Pull request by brainstorm**
Hello @arq5x, @brentp, @ohofmann,

Here's a proof of concept on integrating Puzzle's genome browser with GEMINI. That should tackle issue 
#678 while the builtin browser gets a maintenance fix.

There is more functionality to puzzle that might be interesting to integrate further with GEMINI, so this pullrequest is very much open for discussion.

Please give it a try and come back to us for any doubts:

@moonso, @robinandeer, @guillermo-carrasco

### CODE REVIEW

**Comment by arq5x** [comment]
Thanks for this! We will have a look through it and get back with comments, etc. soon.

File: gemini/gemini_browser.py [hunk 1]
```diff
@@ -232,14 +242,45 @@ def db_schema():
...
+
+def browser_main(parser, args):
+    global database
+    database = args.db
+    browser = args.use
+
+    try:
+        if args.use == "puzzle":
+            browser_puzzle(args)
+        # XXX: https://github.com/dgaston/kvasir
```
Comment (brainstorm) [inline gemini/gemini_browser.py]: @dgaston, feel free to add support for your genome browser over here ;)

Comment (dgaston) [inline gemini/gemini_browser.py]: Thanks, I'll take a look into it. Mine is actually a fairly large application on its own that interfaces just through the GEMINI API. I'll have to dust it off and integrate over the next few weeks

**Comment by brentp** [comment]
Sweet!
@brainstorm could you remove puzzle from the requirements and just wrap in a try/except ImportError with a message to pip install puzzle on failure? We have enough problems with dependencies so I don't want to have additional deps for an optional feature.

Also, I have nearly resurrected the gemini browser and will put it as the default, but you can leave the PR unchanged in that regard.

**Comment by brainstorm** [comment]
Thanks for the feedback @brentp! If I leave the PR unchanged in that regard, puzzle will be the default browser for now (see argparse "default"):

```
    parser_browser.add_argument('--use', metavar='use', default='puzzle',
            help='Which browser to use: builtin or puzzle')
```

I thought that this was the most reasonable since the builtin one is broken, but happy to adapt to whatever you guys choose... even happier if that default browser is indeed puzzle, of course ;P

**Comment by brentp** [comment]
cheers!

**Comment by brentp** [comment]
I have restored the browser functionality that was built in to gemini and made it the default. I also make the import error messages more informative for puzzle as it could be that, e.g. puzzle is installed but flask is not which would also lead to the same message to the user to install puzzle.

thanks for getting this going!

**Pull request merged**